### PR TITLE
Fix return value of openssl_read (infinite loop)

### PR DESCRIPTION
**Pull request by cschlack**
openssl_read should return -1 in case of error.

SSL_read returns values <= 0 in case of error.

A return value of 0 can lead to an infinite loop, so the return value
of ssl_set_error will be returned if SSL_read is not successful (analog
to openssl_write).

### CODE REVIEW

File: src/openssl_stream.c [hunk 1]
```diff
@@ -522,8 +522,9 @@ ssize_t openssl_read(git_stream *stream, void *data, size_t len)
 	openssl_stream *st = (openssl_stream *) stream;
 	int ret;
 
-	if ((ret = SSL_read(st->ssl, data, len)) <= 0)
-		ssl_set_error(st->ssl, ret);
+	if ((ret = SSL_read(st->ssl, data, len)) <= 0) {
+		return ssl_set_error(st->ssl, ret);
+	}
```
Comment (carlosmn) [inline src/openssl_stream.c]: Please stick to the surrounding style: no braces here

**Comment by carlosmn** [comment]
Nice catch, thanks!

File: src/openssl_stream.c (hunk 1 above)
Comment (cschlack) [inline src/openssl_stream.c]: Actually I did copy the style from openssl_write.

Comment (carlosmn) [inline src/openssl_stream.c]: That's an odd one out; the style of the library is not to use branches, and _this function_ did not use braces

**Comment by stmllr** [comment]
Looks like the build on Environment: GENERATOR=Visual Studio 11 Win64, ARCH=64 is broken on master: https://ci.appveyor.com/project/libgit2/libgit2/branch/master

**Comment by ethomson** [comment]
> Looks like the build on Environment: GENERATOR=Visual Studio 11 Win64, ARCH=64 is broken on master: https://ci.appveyor.com/project/libgit2/libgit2/branch/master

The network tests will occasionally fail as networks and target systems are imperfect, as you might expect.

**Comment by carlosmn** [comment]
It's not the network but threading tests. They occasionally fail as they don't make use of the promises the library itself makes.

**Comment by ethomson** [comment]
Thanks!

**Pull request merged**